roxie/ccd/ccdserver.cpp

```diff
@@ -11835,10 +10994,25 @@ class CRoxieServerJoinActivity : public CRoxieServerTwoInputActivity
             createDefaultLeft();
         if ((leftOuterJoin && (activityKind==TAKjoin || activityKind==TAKjoinlight || activityKind==TAKdenormalizegroup)) || limitOnFail)
             createDefaultRight();
-        if ((helper.getJoinFlags() & JFlimitedprefixjoin) && helper.getJoinLimit()) 
+        bool isStable = (helper.getJoinFlags() & JFunstable) == 0;
+        RoxieSortAlgorithm sortAlgorithm;
+        if (forceSpill)
+            sortAlgorithm = isStable ? stableSpillingQuickSortAlgorithm : spillingQuickSortAlgorithm;
+        else
+            sortAlgorithm = isStable ? stableQuickSortAlgorithm : quickSortAlgorithm;
+        if (helper.isLeftAlreadySorted())
+            groupedLeft.setown(createGroupedInputReader(input, NULL)); // This is something of a no-op but helps keep the code simple (IInputBase is not linkable)
```

Comment:
Could be. I could at least add a function createPassthroughGroupedInputReader() to make the calling code clear in its intent.